README.md

```diff
@@ -23,396 +23,345 @@ Stable release versions are manually released from the main branch.
+  stakingParamsVersion,
+  stakingTx,
+  unsignedUnbondingTx,
+  covenantUnbondingSignatures
+})
 ```
 
-### Create slashing transaction
+#### Withdraw
 
-The slashing transaction is the transaction that is sent to Bitcoin in the
-event of the finality provider in which the stake has been delegated to
-performs an offence.
+There are 3 different types of withdrawal transactions:
+1. Withdraw from early unbonding (`createSignedBtcWithdrawEarlyUnbondedTransaction`)
+   - Used when the unbonding period has passed
+2. Withdraw from expired timelock (`createSignedBtcWithdrawStakingExpiredTransaction`)
+   - Used when the staking period has naturally ended
+3. Withdraw from slashed stake (`createSignedBtcWithdrawSlashingTransaction`)
+   - Used when withdrawing slashed funds after timelock expiry
 
-**For the initial phase of the mainnet, there will be no slashing, so the
-following instructions can be safely ignored and are put here for
-completeness.**
+All withdrawal transactions will direct the change balance to the staker's 
+address (provided via `stakerInfo`).
+For more customized transaction options, please refer to the [advanced usage documentation](docs/advanced-btc-tx.md).
 
-First, collect the parameters related to slashing.
-These are Babylon parameters and should be collected from the Babylon system.
+The required input transaction varies depending on the withdrawal method:
+- Early unbonding withdrawal requires the unbonding transaction
+- Timelock expiry withdrawal requires the staking transaction
+- Slashed stake withdrawal requires the slashing transaction
+
+These transactions can be retrieved from either the Babylon Genesis Chain or the Backend API.
 
 ```ts
-// The address to which the slashed funds should go to.
-const slashingAddress: string = "";
-// The slashing percentage rate. It shall be decimal number between 0-1
-const slashingRate: number = 0;
-// The required fee for the slashing transaction in satoshis.
-const minimumSlashingFee: number = 500;
+import {
+  BabylonBtcStakingManager,
+} from "@babylonlabs-io/btc-staking-ts";
+
+const manager = new BabylonBtcStakingManager(
+  btcNetwork,
+  stakingParams,
+  btcProvider,
+  bbnProvider,
+);
+
+// 1. Withdraw from early unbonding (when unbonding period has passed)
+const signedWithdrawEarlyUnbondedTx = await manager.createSignedBtcWithdrawEarlyUnbondedTransaction({
+  stakerInfo,
+  stakingInput,
+  stakingParamsVersion,
+  unbondingTx,
+  feeRate
+})
+
+// 2. Withdraw from expired timelock (when staking period has naturally ended)
+const signedWithdrawTimelockExpiredTx = await manager.createSignedBtcWithdrawStakingExpiredTransaction({
+  stakerInfo,
+  stakingInput,
+  stakingParamsVersion,
+  stakingTx,
+  feeRate
+})
+
+// 3. Withdraw from slashed delegation (after timelock expiry)
+const signedWithdrawSlashedTx = await manager.createSignedBtcWithdrawSlashingTransaction({
+  stakerInfo,
+  stakingInput,
+  stakingParamsVersion,
+  slashingTx,
+  feeRate
+})
 ```
 
```

Comment:
The document does not detail that the user is creating/signing other transactions apart from staking (e.g. slashing) and what their signature means. Also what about PoP?

Reply:
All of those (including the pop) require the caller of the library to invoke a signer for them. They should know what each signature they need to add and what it means